inginious/frontend/webapp/static/js/studio.js

```diff
@@ -487,6 +493,296 @@ function studio_init_template_custom(well, pid, problem)
+    } else { // default value for zoom wheel
+        $("#zoomWheel-" + pid).prop('checked', false);
+    }
+    if ("startScale" in zoom_options) {
+        $("#zoomStartScale-" + pid).val(zoom_options["startScale"]);
+    } else { // default value for start scale
+        $("#zoomStartScale-" + pid).val("1.0");
+    }
+    if ("maxScale" in zoom_options) {
+        $("#zoomMaxScale-" + pid).val(zoom_options["maxScale"]);
+    } else { // default value for max scale
+        $("#zoomMaxScale-" + pid).val("3.0");
+    }
+    if ("minScale" in zoom_options) {
+        $("#zoomMinScale-" + pid).val(zoom_options["minScale"]);
+    } else { // default value for min scale
+        $("#zoomMinScale-" + pid).val("0.3");
+    }
+    if ("scaleSpeed" in zoom_options) {
+        $("#zoomScaleSpeed-" + pid).val(zoom_options["scaleSpeed"]);
+    } else { // default value for zoom
+        $("#zoomScaleSpeed-" + pid).val("1.2");
+    }
+
+    $('#zoom-' + pid).change(function() {
+        if ($(this).is(":checked")) {
+            var row = $("#subproblem_blockly_zoom").html();
+            var new_row_content = row.replace(/PID/g, pid);
+            $(this).parent().parent().parent().append(new_row_content);
+        } else {
+            $(this).parent().parent().parent().html($(this).parent().parent().detach());
+        }
+    });
+
+    if ("files" in problem) {
+        jQuery.each(problem["files"], function(index, elem)
+        {
+            studio_create_javascript_files(pid, elem);
+        });
+    }
+    if ("blocks_files" in problem) {
+        jQuery.each(problem["blocks_files"], function(index, elem)
+        {
+            studio_create_blocks_files(pid, elem);
+        });
+    }
+
+    var factoryController;
+    $('#blockFactoryModal').on('shown.bs.modal', function() {
+        // FIXME Next line is a fix, please monitor https://github.com/google/blockly/issues/56
+        $(document).off('focusin.modal');
+
+        $("#blockFactoryModalBody").html($("#subproblem_blockly_factory").detach());
+
+        var basicToolbox = Toolbox.FULL;
+        var toolbox = $("#toolbox-code").text();
+        var workspaceBlocks = $("#workspace-code").text();
+
+        factoryController = new FactoryController('blocklyFactory', basicToolbox, problem["toolbox"], workspaceBlocks, problem["options"]);
+        var preview = new Preview(factoryController);
+        factoryController.setPreview(preview);
+        factoryController.injectWorkspaces();
+    });
+
+    $('#blockFactoryModal').on('hidden.bs.modal', function() {
+        // If you want this behavior to be BEFORE the visual part, use hide.bs.modal, if after use hidden.bs.modal
+        // When the modal is closed, the workspace (from the initial page must be updated)
+        factoryController.dispose();
+    });
+
+}
+
+/**
+ * Create a new input for the name of a file
+ * @param pid
+ * @param filename
+ */
+function studio_create_javascript_files(pid, filename)
```

Comment:
Function name should reference blockly, as the function only serves this purpose